rai/show.go

```diff
@@ -127,3 +127,42 @@ func (tx *TransactionResult) Show() {
 		}
 	}
 }
+
+func zip(lists ...[]interface{}) func() []interface{} {
+	zip := make([]interface{}, len(lists))
+	i := 0
+	return func() []interface{} {
+		for j := range lists {
+			if i >= len(lists[j]) {
+				return nil
+			}
+			zip[j] = lists[j][i]
+		}
+		i++
+		return zip
+	}
+}
+
+func (tx *TransactionAsyncResult) Show() {
+	res := make(map[string][][]interface{})
+
+	for _, r := range tx.Results {
+		res[r.RelationID] = append(res[r.RelationID], r.Table)
+	}
+
+	for k, v := range res {
+		fmt.Println(k)
+		iter := zip(v...)
+		for tuples := iter(); tuples != nil; tuples = iter() {
+			for i, tuple := range tuples {
+				if i > 0 {
+					fmt.Print(", ")
+				}
+
+				fmt.Print(tuple)
+			}
+			fmt.Println()
+		}
+		fmt.Println()
+	}
+}
```

Comment:
Awesome! Looks great!

Would it be possible to add a unit test of this? Is there a standard way to unit test printing in Go?
If not, could you at least test a few variants just to make sure that it doesn't raise any errors?

Reply:
I think the approach would be to make two variants:
1. `func (tx *TransactionAsyncResult) Write(out io.Writer)` // writes to the given buffer, which can be a string buffer used for testing
2. `func (tx *TransactionAsyncResult) rai.Print()` // just calls `tx.Write(os.Stdout)`

Maybe a similar pattern exists in the Julia SDK? 🤔

Reply:
Ah, yeah, great point. Thanks @vilterp. You're exactly right:
https://github.com/RelationalAI/rai-sdk-julia/blob/6c9ddf55fe791d82b15c8a9e9f80c0bcdec04064/test/api.jl#L150-L153

Although not very extensive tests 😅